### PR TITLE
chore: Refactor docker-compose for production like env

### LINKED ISSUE #5510: Create Production specific dock-compose setup
create  docker-compose suitable for production environment and also there is need of hiding all secrets in a file that will be environment specific

**Approach**
will add all credentials  ie   `(username, passwords)` to `.env` that will be environment specific 

i think it will be the best approach feel free to share  better solution down here 

reffering to task in #5454 

- **working on it ?** 
yes

### PR DESCRIPTION
**Pull request by chetanyachopra**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Changes proposed in 
- docker-compose.yml
- .env.example

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+
+REDIS_CONTAINER_NAME=opev-redis
+REDIS_PORT=6379
+REDIS_URL=redis:6379
+
+APP_CONFIG=config.DevelopmentConfig
+
+WEB_CONTAINER_NAME=opev-web
+
+CELERY_CONTAINER_NAME=opev-celery
```
[the remark this answers is not in the document]
Comment (chetanyachopra) [reply .env.example]: oh sorry my bad !! i'll  update the appConfig ... 
and just one thing should i make a new docker-compose-prod.yml or just refactor in this docker-compose only

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply .env.example]: You should just refactor docker-compose, but don't touch .env.example

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply .env.example]: Learn about docker compose secrets

**Comment by thanostx** [comment]
It would be to wise to define SERVER_NAME for production, as it currently points to open-event-api.herokuapp.com API.

Also why Celery needs root permission to run (C_FORCE_ROOT=true) ?

**Comment by CosmicCoder96** [comment]
@chetanyachopra Any updates on this?

**Comment by chetanyachopra** [comment]
> @chetanyachopra Any updates on this?


i am stuck on how to  approach to solve this !!! there are many ways can you guide me how to approach this 
i tried different approaches and seeking for the guidence from community !!

-  tried with creating secrets with docker compose but as for creating secrets node should be swarm manager or connected to a swarm
 `$ printf 'openv_pass' |  docker secret create postgres_password - ` 
`Error response from daemon: This node is not a swarm manager. Use "docker swarm init" or "docker swarm join" to connect this node to swarm and try again.`
and after creating it a swarm manager i have able to run all the containers in a cluster so the question is are we using swarm for deployment

- another approach i had in mind that can i can put it (credentials) in `.env` and `.env` is also in gitignore by adding it in `.env.example` by doing this credentials are available in compose file and we can just edit  it in  production with the credentials of pproduction environment in `.env` but personally i dont think its the right approach 

- another approach is that we can use kubernets secrets

do you have an idea on how to approach this ! or can you suggest me with any other better solution you have in mind !! @CosmicCoder96 . I'll consider that and update as soon as possible

**Comment by iamareebjamal** [comment]
You don't have to use swarm secrets. Just specify an env file like web.env and add it in .gitignore

File: .env.example [hunk 2]
```diff
@@ -1,6 +1,10 @@
 DATABASE_URL=postgresql://john:start@127.0.0.1:5432/oevent
 INTEGRATE_SOCKETIO=false
 TEST_DATABASE_URL=postgresql://open_event_user:test@127.0.0.1:5432/opev_test
-APP_CONFIG=config.DevelopmentConfig
+APP_CONFIG=config.ProductionConfig
```
Comment (iamareebjamal) [inline .env.example]: Don't change this

**Comment by chetanyachopra** [comment]
please review @iamareebjamal

**Comment by iamareebjamal** [comment]
Will test this

**Comment by chetanyachopra** [comment]
> Will test this

cool :+1: @iamareebjamal   just for your reference this is output of my docker-compose config  which is same as before and just confirms  that variables are accessable inside the docker compose yml file 

`$docker-compose config`
```json
services:
  celery:
    build:
      context: /home/cheti/Desktop/fossasia/open-event-server
    container_name: opev-celery
    depends_on:
    - elastic
    - postgres
    - redis
    environment:
      ADMIN_EMAIL: '@{ADMIN_EMAIL}'
      ADMIN_PASSWORD: '@{ADMIN_PASSWORD}'
      C_FORCE_ROOT: "true"
      DATABASE_URL: postgresql://open_event_user:opev_pass@postgres:5432/open_event
      DEPLOYMENT: celery
      ELASTICSEARCH_HOST: elastic:9200
      REDIS_URL: redis://redis:6379/0
    links:
    - postgres:postgres
    - redis:redis
    - elastic:elastic
    restart: unless-stopped
    volumes:
    - /home/cheti/Desktop/fossasia/open-event-server/static:/data/app/static:rw
  elastic:
    container_name: opev-elastic-search
    environment:
      discovery.type: single-node
    image: elasticsearch:5.6.12-alpine
    ports:
    - published: 9200
      target: 9200
    - published: 9300
      target: 9300
    restart: unless-stopped
    volumes:
    - es:/usr/share/elasticsearch/data:rw
  postgres:
    container_name: opev-postgres
    environment:
      POSTGRES_DB: open_event
      POSTGRES_PASSWORD: opev_pass
      POSTGRES_USER: open_event_user
    image: postgres:10-alpine
    restart: unless-stopped
    volumes:
    - pg:/var/lib/postgresql/data:rw
  redis:
    command: redis-server
    container_name: opev-redis
    image: redis:3-alpine
    restart: unless-stopped
    volumes:
    - rd:/var/lib/redis/data:rw
  web:
    build:
      context: /home/cheti/Desktop/fossasia/open-event-server
    container_name: opev-web
    depends_on:
    - elastic
    - postgres
    - redis
    environment:
      ADMIN_EMAIL: '@{ADMIN_EMAIL}'
      ADMIN_PASSWORD: '@{ADMIN_PASSWORD}'
      DATABASE_URL: postgresql://open_event_user:opev_pass@postgres:5432/open_event
      ELASTICSEARCH_HOST: elastic:9200
      REDIS_URL: redis://redis:6379/0
    links:
    - postgres:postgres
    - redis:redis
    - elastic:elastic
    ports:
    - published: 8080
      target: 8080
    restart: unless-stopped
    volumes:
    - /home/cheti/Desktop/fossasia/open-event-server/static:/data/app/static:rw
version: '3.5'
volumes:
  es: {}
  pg: {}
  rd: {}
 ```

**Comment by chetanyachopra** [comment]
hi @iamareebjamal looking forward for your response on this one

**Comment by iamareebjamal** [comment]
/remind me to review this in 2 days

**Comment by chetanyachopra** [comment]
can we merge these changes ?? @iamareebjamal

**Pull request merged**